drl/nn/nn.py

```diff
@@ -0,0 +1,453 @@
+				'blue')
+			names = ["Rectified Linear:", "Rectified Linear 6:", "Concatenated ReLU:", "Exponential Linear:",
+			         "Softplus:", "Softsign:", "Sigmoid:", "Tanh:", "Linear:"]
+			codes = ["'relu'", "'relu6'", "'crelu'", "'elu'", "'softplus'", "'softsign'", "'sigmoid'", "'tanh'",
+			         "'linear'"]
+			headers = ["Name", "Code"]
+			print_table([names, codes], headers, indentation='center', color='blue')
+			exit()
+
+	def pooling(self, input_layer, name=None):
+		"""
+		Applies the type of pooling as selected by the user. The default pooling type used is max_pooling.
+
+			:param input_layer: Layer that is fed as input for applying pooling
+			:param name: Optional Parameter. Allows you to name the layer.
+			:return: Pooling applied to the input layer.
+		"""
+		if self.options['pooling_type'] == 'max_pooling':
+			return tf.nn.max_pool(input_layer,
+			                      ksize=[1, self.options['pooling_ksize'], self.options['pooling_ksize'], 1],
+			                      strides=[1, self.options['pooling_stride'], self.options['pooling_stride'], 1],
+			                      padding='SAME', name=name)
+		elif self.options['pooling_type'] == 'avg_pooling':
+			return tf.nn.avg_pool(input_layer,
+			                      ksize=[1, self.options['pooling_ksize'], self.options['pooling_ksize'], 1],
+			                      strides=[1, self.options['pooling_stride'], self.options['pooling_stride'], 1],
+			                      padding='SAME', name=name)
+		else:
+			color_print("ERROR: Please select a pooling operation that is supported by this library!", color='red',
+			            mode='bold')
+			print()
+			color_print(
+				"This library supports the following pooling functions. Please use the correct code. Make sure there are no spelling mistakes!",
+				'blue')
+			names = ["Max Pooling:", "Average Pooling:"]
+			codes = ["'max_pooling'", "'avg_pooling'"]
+			headers = ["Name", "Code"]
+			print_table([names, codes], headers, indentation='center', color='blue')
+			exit()
+
+	def batch_norm(self, input_layer, decay=0.999, epsilon=1e-3, name=None):
+		"""
+		Batch Normalization wrapper.
+		Original implementation: https://r2rt.com/implementing-batch-normalization-in-tensorflow.html
+
+			:param input_layer: Input layer to which we will apply BN
+			:param decay: Decay for the moving average. Reasonable values for decay are close to 1.0.
+			:param epsilon: Small float added to variance to avoid dividing by zero.
+			:param name: Optional Parameter. Allows you to name the layer.
+			:return: Batch normalized layer output.
+		"""
+		scale = tf.Variable(tf.ones([input_layer.get_shape()[-1]]))
+		beta = tf.Variable(tf.zeros([input_layer.get_shape()[-1]]))
+		pop_mean = tf.Variable(tf.zeros([input_layer.get_shape()[-1]]), trainable=False)
+		pop_var = tf.Variable(tf.ones([input_layer.get_shape()[-1]]), trainable=False)
+
+		if self.is_training:
```

Comment:
You cannot use a tf.placeholder(bool) in an if statement. You have to implement like [this](https://github.com/BartKeulen/drl/blob/feature/ddpgSmartStart/drl/utilities/tfnetwork.py#L117)